### PR TITLE
[Logs SDK] LogProcessor, LogExporter changes

### LINKED ISSUE #1694: [Logs SDK] LogProcessor, LogExporter changes
specs 
- https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/sdk.md#logrecordprocessor
- https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/sdk.md#logrecordexporter

- Rename `LogProcessor` to `LogRecordProcessor` 
- Rename `LogExporter` to `LogRecordExporter`

### PR DESCRIPTION
**Pull request by owent**
Signed-off-by: owent <admin@owent.net>

Fixes #1694 

## Changes

**These are break changes.**

+ Rename `LogProcessor` to `LogRecordProcessor`
+ Rename `LogExporter` to `LogRecordExporter`
+ Move `*log_processor*` to `*log_record_processor*`
+ Move `*log_exporter*` to `*log_record_exporter*`

For significant contributions please make sure you have completed the following items:

* [x] `CHANGELOG.md` updated for non-trivial changes
* [x] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by owent** [comment]
Is it possiable to review and merge this PR first, before other changes.
There are a lot of renames in this PR, and it's easy to conflict with others.
@open-telemetry/cpp-approvers

**Review by marcalff — APPROVED** [review]
LGTM

**Review by esigo — APPROVED** [review]
LGTM
Thanks for the fix :)

**Comment by lalitb** [comment]
One minor comment - Should we be changing all the exporters class names to use "*LogRecordExporter", to be in consistent with the file-names, and their cmake/bazel targets?

**Comment by esigo** [comment]
> One minor comment - Should we be changing all the exporters class names to use "*LogRecordExporter", to be in consistent with the file-names, and naming of their cmake/bazel targets?
> 
> So, `ElasticsearchLogExporter` to `ElasticsearchLogRecordExporter` `OStreamLogExporter` to `OStreamLogRecordExporter` `OtlpGrpcLogExporter` to `OtlpGrpcLogRecordExporter` `OtlpHttpLogExporter` to `OtlpHttpLogRecordExporter`
> 
> And similarly for Processors: `BatchLogProcessor` to `BarchLogRecordProcessor` `SimpleLogProcessor` to `SimpleLogRecordProcessor`

yeah, I can do it tomorrow.

File: sdk/src/logs/multi_log_record_processor_factory.cc [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#ifdef ENABLE_LOGS_PREVIEW
```
Comment (lalitb) [inline sdk/src/logs/multi_log_record_processor_factory.cc:4]: Not related to this PR, but I think we can later remove the factory class for MultiLog<Record>Processor, as this class is only used internally in SDK.

**Pull request merged**